sdk/tests/iot/common/test_az_iot_common.c

```diff
@@ -258,6 +258,22 @@ static void test_az_iot_provisioning_client_logging_succeed()
   az_log_set_classifications(NULL);
 }
 
+#ifndef AZ_NO_LOGGING
+static void test_az_iot_calculate_retry_delay_no_logging_succeed()
+{
+  az_log_classification const classifications[] = { AZ_LOG_END_OF_LIST };
+  az_log_set_classifications(classifications);
+  az_log_set_callback(_log_listener);
+
+  _log_retry = 1;
+  assert_int_equal(2229, az_iot_calculate_retry_delay(5, 1, 500, 100000, 1234));
+  assert_int_equal(_az_BUILT_WITH_LOGGING(1, 0), _log_retry);
```

Comment:
I _think_ you'd need to make one of these two changes:
```suggestion
  _log_retry = 0;
  assert_int_equal(2229, az_iot_calculate_retry_delay(5, 1, 500, 100000, 1234));
  assert_int_equal(_az_BUILT_WITH_LOGGING(0, 0), _log_retry);
```

OR

```suggestion
  _log_retry = 1;
  assert_int_equal(2229, az_iot_calculate_retry_delay(5, 1, 500, 100000, 1234));
  assert_int_equal(_az_BUILT_WITH_LOGGING(1, 1), _log_retry);
```

If logging isn't on, `_log_retry` should not change and the callback method (`_log_listener`) shouldn't get called. I would go one step further and add a log listener with just `assert(false)` to validate that:
https://github.com/Azure/azure-sdk-for-c/blob/96ab13b52cbf1186bc64965c166cc19a2a805640/sdk/tests/core/test_az_logging.c#L277-L282

Reply:
ahhh I see how this thing works now. I misread it will update

Reply:
updated. thanks @ahsonkhan